internal/api/incident.go

```diff
@@ -0,0 +1,125 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/tidwall/gjson"
+)
+
+// Incident is a type of ticket
+type Incident struct {
+	Cluster     string    `json:"cluster,omitempty"`
+	Component   string    `json:"component,omitempty"`
+	Comments    []Comment `json:"comments,omitempty"`
+	Description string    `json:"description,omitempty"`
+	Identifier  string    `json:"external_identifier,omitempty"`
+	IssueID     string    `json:"issue_id,omitempty"`
+	Priority    string    `json:"priority,omitempty"`
+	Status      string    `json:"status,omitempty"`
+	Summary     string    `json:"summary,omitempty"`
+}
+
+// Comment is an individual post
+type Comment struct {
+	ID     string `json:"id,omitempty"`
+	Author string `json:"owner,omitempty"`
+	Body   string `json:"body,omitempty"`
+}
+
+// incidentUpdate is an event
+type incidentUpdate struct {
+	incident *Incident
+	sqs      Messenger
+}
+
+// NewIncident initialises an Incident
+func NewIncident() *Incident {
+	return &Incident{}
+}
+
+// checkVars checks incoming payload has necessary fields
+func checkVars(input string) error {
+
+	vars := []string{
+		"COMPONENT_FIELD",
```

Comment:
are there some fields not required then, such as `CLUSTER_FIELD`?

Reply:
yes, cluster field is optional on JSD form so may not always be provided. all field mappings are currently provisional. onec we're assigned a BA, we'll have an opportunity to revise/match fields

Reply:
Great thanks